test/fail_compilation/bug15613.d

```diff
@@ -0,0 +1,18 @@
+/*
+TEST_OUTPUT:
+---
+fail_compilation/bug15613.d(16): Error: function `bug15613.f(int...)` is not callable using argument types `(typeof(null))`
+fail_compilation/bug15613.d(16):        cannot pass argument `null` of type `typeof(null)` to parameter `int...`
```

Comment:
I suggest special casing `null` so it reads:  "cannot pass `null` to parameter `int...`".  "`null` of type `typeof(null)` is quite an odd thing to say.  I'll still approve this PR without this change, as this PR is a huge improvement over the current implementation, so it's your call.

Reply:
Ideally, maybe. This is a general issue though, e.g. `int x = null;` gives:

    funcvar.d(5): Error: cannot implicitly convert expression `null` of type `typeof(null)` to `int`